src/test/java/bio/terra/service/filedata/azure/tables/TableDaoConnectedTest.java

```diff
@@ -184,4 +192,97 @@ private void checkThatEntriesExist(UUID collectionId, String tableName, boolean
+
+    // And some files that won't be in the snapshot
+    List<FireStoreDirectoryEntry> dsetObjects = new ArrayList<>();
+    dsetObjects.add(makeFileObject(datasetId, "/adir/bdir/B2", 16));
+    dsetObjects.add(makeFileObject(datasetId, "/adir/A2", 32));
+
+    // Make the dataset file system
+    List<FireStoreDirectoryEntry> fileObjects = new ArrayList<>(snapObjects);
+    fileObjects.addAll(dsetObjects);
+    for (FireStoreDirectoryEntry fireStoreDirectoryEntry : fileObjects) {
+      tableDirectoryDao.createDirectoryEntry(
+          tableServiceClient, datasetId, "dataset", fireStoreDirectoryEntry);
+    }
+
+    // Make the snapshot file system
+    List<String> fileIdList = new ArrayList<>();
+    for (FireStoreDirectoryEntry fireStoreDirectoryEntry : snapObjects) {
+      fileIdList.add(fireStoreDirectoryEntry.getFileId());
+    }
+    tableDirectoryDao.addEntriesToSnapshot(
+        tableServiceClient, tableServiceClient, datasetId, "dataset", snapshotId, fileIdList);
+
+    // Validate we cannot lookup dataset files in the snapshot
+    for (FireStoreDirectoryEntry dsetObject : dsetObjects) {
+      FireStoreDirectoryEntry snapObject =
+          tableDirectoryDao.retrieveById(
+              tableServiceClient,
+              StorageTableName.SNAPSHOT.toTableName(snapshotId),
+              dsetObject.getFileId());
+      assertNull("object not found in snapshot", snapObject);
+    }
+
+    // Compute the size and checksums
+    FireStoreDirectoryEntry topDir =
+        tableDirectoryDao.retrieveByPath(tableServiceClient, datasetId, "dataset", "/");
+    tableDao.snapshotCompute(snapshot, tableServiceClient, tableServiceClient);
+
+    // Check the accumulated size on the root dir
+    FireStoreDirectoryEntry snapObject =
+        tableDirectoryDao.retrieveByPath(
+            tableServiceClient, snapshotId, StorageTableName.SNAPSHOT.toTableName(snapshotId), "/");
+    assertNotNull("root exists", snapObject);
+    assertThat("Total size is correct", snapObject.getSize(), equalTo(15L));
+    assertThat(
+        "The directory had its checksum calculated",
+        snapObject.getChecksumMd5(),
+        equalTo("52392aba5378fddebf71225639606ead"));
+  }
+
+  private FireStoreDirectoryEntry makeFileObject(UUID datasetId, String fullPath, long size) {
+
+    String fileId = UUID.randomUUID().toString();
+
+    FireStoreFile newFile =
+        new FireStoreFile()
+            .fileId(fileId)
+            .mimeType("application/test")
+            .description("test")
+            .bucketResourceId("test")
+            .fileCreatedDate(Instant.now().toString())
+            .gspath(endpoint + "/" + fullPath)
+            .checksumMd5(VirtualFileSystemUtils.computeMd5(fullPath))
+            .size(size);
+
+    tableFileDao.createFileMetadata(tableServiceClient, newFile);
+
+    return new FireStoreDirectoryEntry()
+        .fileId(fileId)
+        .isFileRef(true)
+        .path(FileMetadataUtils.getDirectoryPath(fullPath))
```

Comment:
The row_id for directory entries is based on the file path (and we currently re-use the same table for datasets) so we should add a random uuid to the path per test-run for now to avoid conflicts with tests running in parallel!

Reply:
Oh the partition key actually includes the dataset-id so the records won't conflict! I think it's still okay to keep the unique directories in though since you already made the change